src/main/java/com/khoros/twitterapp/services/CacheUp.java

```diff
@@ -2,20 +2,31 @@
 
 import twitter4j.Status;
 
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
 public class CacheUp {
-    // clean up interval 1 week
-    private Set<Status> cacheSet = new HashSet<>();
+    private HashMap<TwitterService.CacheSetType, Set<Status>> cacheMap = new HashMap<>();
 
-    public Set<Status> getCacheSet() {
-        return cacheSet;
+    public CacheUp() {
+        cacheMap.put(TwitterService.CacheSetType.HOME, new HashSet<>());
+        cacheMap.put(TwitterService.CacheSetType.USER, new HashSet<>());
     }
 
-    public void addStatusesToCache(List<Status> statusList) {
-        statusList.forEach(status -> cacheSet.add(status));
+    public Set<Status> getTimelineCache(TwitterService.CacheSetType setType) {
+        System.out.println(setType);
+        return cacheMap.get(setType);
+    }
+
+    public void addStatusToCache(TwitterService.CacheSetType setType, List<Status> statusList) {
+
+        Set<Status> currentSet = cacheMap.get(setType);
+
+        statusList.forEach(status -> currentSet.add(status));
```

Comment:
why the need to loop and add?

Reply:
I didn't notice the addAll method the sets inherited in the docs